src/sinks/syslog.rs

```diff
@@ -0,0 +1,594 @@
+}
+
+fn default_severity() -> SyslogSeverity {
+    SyslogSeverity::SEV_INFO
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(tag = "mode", rename_all = "snake_case")]
+pub enum Mode {
+    Tcp(TcpSinkConfig),
+    Udp(UdpSinkConfig),
+    #[cfg(unix)]
+    Unix(UnixSinkConfig),
+}
+
+inventory::submit! {
+    SinkDescription::new::<SyslogSinkConfig>("syslog")
+}
+
+impl GenerateConfig for SyslogSinkConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(
+            r#"address = "2001:db8::1:514"
+            mode = "tcp"
+            "#,
+        )
+        .unwrap()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "syslog")]
+impl SinkConfig for SyslogSinkConfig {
+    async fn build(
+        &self,
+        cx: SinkContext,
+    ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
+        let format = self.format;
+        let include_extra_fields = self.include_extra_fields;
+        let appname_key = self.appname_key.to_owned();
+        let facility_key = self.facility_key.to_owned();
+        let host_key = self.host_key.to_owned();
+        let msgid_key = self.procid_key.to_owned();
+        let procid_key = self.msgid_key.to_owned();
+        let severity_key = self.severity_key.to_owned();
+        let default_facility = self.default_facility;
+        let default_severity = self.default_severity;
+
+        let syslog_encode = move |event, include_len| {
+            build_syslog_message(
+                event,
+                include_len,
+                format,
+                include_extra_fields,
+                appname_key.as_str(),
+                facility_key.as_str(),
+                host_key.as_str(),
+                msgid_key.as_str(),
+                procid_key.as_str(),
+                severity_key.as_str(),
+                default_facility,
+                default_severity,
+            )
+        };
+
+        match &self.mode {
+            Mode::Tcp(config) => config.build(cx, move |e| Some(syslog_encode(e, true))),
+            Mode::Udp(config) => config.build(cx, move |e| Some(syslog_encode(e, false))),
+            #[cfg(unix)]
+            Mode::Unix(config) => config.build(cx, move |e| Some(syslog_encode(e, true))),
+        }
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "syslog"
+    }
+}
+
+fn build_syslog_message(
+    event: Event,
+    include_len: bool,
+    format: Format,
+    include_extra_fields: bool,
+    appname_key: &str,
+    facility_key: &str,
+    host_key: &str,
+    msgid_key: &str,
+    procid_key: &str,
+    severity_key: &str,
+    default_facility: SyslogFacility,
+    default_severity: SyslogSeverity,
+) -> Bytes {
+    let mut log = event.into_log();
+
+    let ts = log.remove(log_schema().timestamp_key()).and_then(|v| {
+        v.as_timestamp()
+            .map(|v| FixedOffset::west(0).timestamp_nanos(v.timestamp_nanos()))
```

Comment:
This seems like this could be handled easier. What is this accomplishing? Is `FixedOffset::west(0)` different than `Utc`?